File: solutions/c/rna-transcription/3/rna_transcription.c
#include "rna_transcription.h"
#include <stdlib.h>

char *to_rna(const char *dna){
    char* rna = malloc(100*sizeof(char));
    rna[0]='\0';
    for (int i = 0; dna[i]; i++) {
        switch (dna[i]) {
            case 'G': rna[i] = 'C'; break;
            case 'C': rna[i] = 'G'; break;
            case 'T': rna[i] = 'A'; break;
            case 'A': rna[i] = 'U'; break;
        }
    }
    return rna;
}

File: solutions/r/rna-transcription/3/rna-transcription.R
to_rna <- function(dna) {
	rna <- ""
	d2r <- list("G"="C","C"="G","T"="A","A"="U")
	for (d in unlist(strsplit(dna, ""))) {
		r <- d2r[[d]]
		if (is.null(r)) { stop() }
		rna <- paste(rna, r, sep="")
		#print (d)
		#print (rna)
	}
	return (rna)
}
